packages/features/bookings/lib/handleNewBooking/test/booking-limits.test.ts

```diff
@@ -1,7 +1,464 @@
+                {
+                  eventTypeId: 1,
+                  userId: 101,
+                  status: BookingStatus.ACCEPTED,
+                  startTime: `${nextYearDateString}T04:30:00.000Z`,
+                  endTime: `${nextYearDateString}T05:00:00.000Z`,
+                },
+              ],
+              organizer,
+            })
+          );
+
+          const mockBookingData = getMockRequestDataForBooking({
+            data: {
+              start: `${nextYearDateString}T05:00:00.000Z`,
+              end: `${nextYearDateString}T05:30:00.000Z`,
+              eventTypeId: 1,
+              responses: {
+                email: booker.email,
+                name: booker.name,
+                location: { optionValue: "", value: "New York" },
+              },
+            },
+          });
+
+          const { req } = createMockNextJsRequest({
+            method: "POST",
+            body: mockBookingData,
+          });
+
+          vi.spyOn(prismock, "$queryRaw").mockResolvedValue([{ totalMinutes: yearlyDurationLimit }]);
+
+          await expect(async () => await handleNewBooking(req)).rejects.toThrowError(
+            "duration_limit_reached"
+          );
+
+          const { dateString: yearWithoutBookingsDateString } = getDate({ yearIncrement: 2 });
+
+          const mockBookingDataFollowingYear = getMockRequestDataForBooking({
+            data: {
+              start: `${yearWithoutBookingsDateString}T05:00:00.000Z`,
+              end: `${yearWithoutBookingsDateString}T05:30:00.000Z`,
+              eventTypeId: 1,
+              responses: {
+                email: booker.email,
+                name: booker.name,
+                location: { optionValue: "", value: "New York" },
+              },
+            },
+          });
+
+          const { req: reqFollowingYear } = createMockNextJsRequest({
+            method: "POST",
+            body: mockBookingDataFollowingYear,
+          });
+
+          vi.spyOn(prismock, "$queryRaw").mockResolvedValue([{ totalMinutes: 0 }]);
+
+          const createdBooking = await handleNewBooking(reqFollowingYear);
+
+          expect(createdBooking.responses).toContain({
+            email: booker.email,
+            name: booker.name,
+          });
+
+          expect(createdBooking).toContain({
+            location: "New York",
+          });
+
+          await expectBookingToBeInDatabase({
+            description: "",
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            uid: createdBooking.uid!,
+            eventTypeId: mockBookingDataFollowingYear.eventTypeId,
+            status: BookingStatus.ACCEPTED,
+          });
+        },
+        timeout
+      );
+
+      const { date: todayDate } = getDate();
+      const { date: tomorrowDate } = getDate({ dateIncrement: 1 });
+
+      // today or tomorrow can't be the 1st day of month for this test to succeed
+      test.skipIf([todayDate, tomorrowDate].includes("01"))(
```

Comment:
This is required as we need to seed bookings in the past while ensuring future bookings are made in the same month.

It's a bit less invasive than mocking server time, but maybe you have a better idea.